### PR TITLE
[KM169] Ignore empty changes, ensure that delete works

### PR DESCRIPTION
**Pull request by paulbes**
Ignore empty changes, ensure that delete works

## Description
The tests will run locally, but not on CI for now, maybe we can make them work on CI also

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

**Review by yngvark — APPROVED** [review]
I like the code :+1: 

Make sure to test that existing records in origo-aws-infrastructure pattern follows any pattern changes here, if there are any (like a new pattern/system for file names for instance). Check out the git log in the  origo-aws-infrastructure repository to see an example of this ("refactor" commits).

File: pkg/git/git_test.go [hunk 1]
```diff
@@ -1,37 +1,48 @@
...
-		expect string
-	}{
-		{
-			name:   "Should work",
-			path:   "../..",
-			expect: "oslokommune/okctl",
+	runner := git.New()
+
+	changeSet := &git.ChangeSet{
+		Stager:        git.RepositoryStagerClone(git.RepositoryURL(github.DefaultOrg, "okctl-repository-test")),
```
Comment (deifyed) [inline pkg/git/git_test.go:25]: Test requiring an internet connection warning

File: pkg/git/git_test.go [hunk 2]
```diff
@@ -1,37 +1,48 @@
...
+		Actions: []git.ActionFn{
+			git.AddFile("", "my.file", []byte("some nice data")),
 		},
 	}
 
-	for _, tc := range testCases {
-		tc := tc
+	result, err := runner.UpdateRepository(changeSet)
+	assert.NoError(t, err)
+	fmt.Println(result)
```
Comment (deifyed) [inline pkg/git/git_test.go:37]: Println's?

File: pkg/git/git_test.go [hunk 3]
```diff
@@ -1,37 +1,48 @@
...
-		})
+	changeSet.CommitMessage = "removing some data"
+	changeSet.Actions = []git.ActionFn{
+		git.RemoveFile("", "my.file"),
 	}
+	changeSet.FileSystem = memfs.New()
+
+	result, err = runner.UpdateRepository(changeSet)
+	assert.NoError(t, err)
+	fmt.Println(result)
```
Comment (deifyed) [inline pkg/git/git_test.go:47]: Println's?

**Review by deifyed — APPROVED** [review]

**Pull request merged**